pytorch_lightning/trainer/data_loading.py

```diff
@@ -252,6 +252,67 @@ def replace_sampler(self, dataloader: DataLoader, sampler, mode: Optional[Runnin
         dataloader = dl_cls(**dl_kwargs)
         return dataloader
 
+    def apply_user_function(self, dataloader: DataLoader, user_function: Callable) -> DataLoader:
```

Comment:
what if we had instead:
```python

def train_dataset(self):
    return Dataset(...)
    
def create_dataloader(self, dataset):
     return DataLoader(dataset, ...)
```
     
then by default we can call `module.create_dataloader(module.train_dataset)` and if needed one can intercept the dataset before passing it to the dataloader creation.

Reply:
Done !